meta-lmp-base/classes/uboot-fitimage.bbclass

```diff
@@ -210,6 +210,30 @@ do_deploy_prepend() {
 		ATF_LOAD_ADDR=$(${READELF} -h ${ATF_ELF} | egrep -m 1 -i "entry point.*?0x" | sed -r 's/.*?(0x.*?)/\1/g')
 	fi
 
+	SECONDARY_OFFSET=`grep 'define CONFIG_SECONDARY_BOOT_SECTOR_OFFSET' ${B}/${config}/u-boot.cfg | cut -d' ' -f 3`
+	if [ "${SECONDARY_OFFSET}" != "0" ]; then
+		bbnote "Building Secondary Image Table, firstSectorNumber = ${SECONDARY_OFFSET}"
+cat << EOF > sit_template
+#! /usr/bin/perl -w
+use strict;
+
+open(my \$out, '>:raw', 'sit.bin') or die "Unable to open: $!";
+print \$out pack("V", 0x0);	# chipnum
+print \$out pack("V", 0x0);	# driver type
+print \$out pack("V", 0x00112233);	# tag
+print \$out pack("V", %sectornum%);	# firstSectorNumber
+print \$out pack("V", 0x0);	# Sector count, not used
+close(\$out);
+
+EOF
+		sed -e s/%sectornum%/${SECONDARY_OFFSET}/g sit_template > sit_gen
+		chmod +x sit_gen
+		./sit_gen
+		objcopy -I binary -O binary --pad-to 512 sit.bin
+		rm sit_gen sit_template
+		install -m 0644 sit.bin ${DEPLOYDIR}/sit-${MACHINE}.bin
+	fi
```

Comment:
I think we should migrate this logic to u-boot-fio-common.inc instead, as the scope of this class is just to generate the fit and perform the required signature.

Reply:
yes, makes sense, will do

I think it also makes sense to remove sit image generation from imx-boot recipe

Reply:
Won't this cause 2 sit.bin to be generated for `imx8*` targets where `imx-boot` is built + u-boot fit-image?

Reply:
@mike-scott I think I'll just remove that logic from imx-boot

Reply:
fixed